TF_Service_and_Tooling_Unity/Assets/Plugins/LeapMotion/Core/Scripts/LeapServiceProvider.cs

```diff
@@ -471,30 +473,110 @@ public enum TrackingOptimizationMode {
       }
       #endif
     }
- 
+    
+    private Coroutine _changeModeCoroutine;
+
     /// <summary>
-    /// Initializes Leap Motion policy flags.
+    /// Sets the initial mode if it is anything other than None, None leave the tracking system in its last state
     /// </summary>
-    protected virtual void initializeFlags() {
-      if (_leapController == null) {
-        return;
+    protected virtual void initialiseMode()
+    {
+      changeMode(_mode);
+    }
+
+    /// <summary>
+    /// Triggers a coroutine that sets appropriate policy flags and wait for them to be set to ensure we've changed mode
+    /// </summary>
+    /// <param name="mode">Mode to set</param>
+    public void changeMode(Mode mode)
+    {
+      if (_changeModeCoroutine != null)
+      {
+        StopCoroutine(_changeModeCoroutine);
+      }
+      
+      switch (mode)
+      {
+        case Mode.None:
+          break;        
+        case Mode.Desktop:
+          _changeModeCoroutine = StartCoroutine(setMode(setDesktopModeFlags, mode));
+          break;
+        case Mode.ScreenTop:
+          _changeModeCoroutine = StartCoroutine(setMode(setScreenTopModeFlags, mode));
+          break;
+        case Mode.HeadMounted:
+          _changeModeCoroutine = StartCoroutine(setMode(setHeadMountedModeFlags, mode));
```

Comment:
Does setting this value when a coroutine is running kill the old coroutine or does it end up running headless? Concerned about having two competing with one another but not really sure how Unity Coroutines work regarding this.

Reply:
This code is handled by XR team but to explain, the coroutine will never be running when a new one is started because it is always stopped at the beginning of this function, before this highlighted code executes

Reply:
Ah yeah I missed the Stop at the start. More for my Unity knowledge than for the sake of the code. I guess then that overwriting a coroutine _doesn't_ automatically stop the original? Good to know for future.

Reply:
Indeed. You don't _have_ to store it as a coroutine, you can start them headlessly and they will just execute, but to handle the stopping of them you must store a reference. (or call StopAllCoroutines()).

As you guessed in your first comment, if we weren't calling StopCoroutine() above, they would all continue executing